docs/source/preparing_datasets/basic_dataset_conversion.md

```diff
@@ -0,0 +1,329 @@
+
+| Category           | Name          | Class        | Notes                    |
+|--------------------|---------------|--------------|--------------------------|
+| Encoding           | 'bytes'       | `Bytes`      | no-op encoding           |
+| Encoding           | 'str'         | `Str`        | stores in UTF-8          |
+| Encoding           | 'int'         | `Int`        | uses `numpy.int64`       |
+| Numpy Array        | 'ndarray'     | `NDArray`    | uses `numpy.ndarray`     |
+| Numpy Unsigned Int | 'uint8'       | `UInt8`      | uses `numpy.uint8`       |
+| Numpy Unsigned Int | 'uint16'      | `UInt16`     | uses `numpy.uint16`      |
+| Numpy Unsigned Int | 'uint32'      | `Uint32`     | uses `numpy.uint32`      |
+| Numpy Unsigned Int | 'uint64'      | `Uint64`     | uses  `numpy.uint64`     |
+| Numpy Signed Int   | 'int8'        | `Int8`       | uses  `numpy.int8`       |
+| Numpy Signed Int   | 'int16'       | `Int16`      | uses  `numpy.int16`      |
+| Numpy Signed Int   | 'int32'       | `Int32`      | uses  `numpy.int32`      |
+| Numpy Signed Int   | 'int64'       | `Int64`      | uses  `numpy.int64`      |
+| Numpy Float        | 'float16'     | `Float16`    | uses  `numpy.float16`    |
+| Numpy Float        | 'float32'     | `Float32`    | uses  `numpy.float32`    |
+| Numpy Float        | 'float64'     | `Float64`    | uses  `numpy.float64`    |
+| Numerical String   | 'str_int'     | `StrInt`     | stores in UTF-8          |
+| Numerical String   | 'str_float'   | `StrFloat`   | stores in UTF-8          |
+| Numerical String   | 'str_decimal' | `StrDecimal` | stores in UTF-8          |
+| Image              | 'pil'         | `PIL`        | raw PIL image            |
+| Image              | 'jpeg'        | `JPEG`       | PIL image as JPEG        |
+| Image              | 'png'         | `PNG`        | PIL image as PNG         |
+| Pickle             | 'pkl'         | `Pickle`     | arbitrary Python objects |
+| JSON               | 'json'        | `JSON`       | arbitrary data as JSON   |
+
+Here's an example where the field `x` is an image, and `y` is a class label, as an integer.
+<!--pytest.mark.skip-->
+```python
+column = {
+    'x': 'jpeg',
+    'y': 'int8'
+}
+```
+
+If the data type you need is not listed in the above table, then you can write your own data type class with `encode` and `decode` methods in it and patch it inside streaming. For example, let's say, you wanted to add an `int32` data type:
+
+<!--pytest.mark.skip-->
+```python
+import numpy as np
+from typing import Any
+
+from streaming.base.format.mds.encodings import Encoding, _encodings
+
+class Int32(Encoding):
+    def encode(self, obj: Any) -> bytes:
+        return obj.tobytes()
```

Comment:
have you run or tested any of this code

```
return np.int32(obj).tobytes()
```

Reply:
Also the np.frombuffer example below returns an array

Reply:
Oh this was already in the documentation so I assumed it was working. Will test it out

Reply:
edited to use `complex128` datatype since we already have int32 lol.